### PR TITLE
fixed several issues to run one test suite successfully. 

### PR DESCRIPTION
**Pull request by rghetia**
- designate span with parent spanId of zero as root span

- added short iteration with expected span count check to reduce the wait time.

- fixed unmarshalling testSuites.

- handled bad reqId from span.

- Removed StartOnce for sender.

### CODE REVIEW

File: interoptest/src/testcoordinator/validator/trace_constructor_test.go [hunk 1]
```diff
@@ -12,67 +12,104 @@ var (
...
+	span22 = &tracepb.Span{TraceId: traceID2[:], SpanId: spanID22[:], ParentSpanId: spanID21[:]} // child
+	span23 = &tracepb.Span{TraceId: traceID2[:], SpanId: spanID22[:]}                            // another root
 )
 
 func TestReconstructTraces(t *testing.T) {
-	got, errs := ReconstructTraces(span3, span4, span1, span5, span2)
-	if len(errs) > 0 {
-		t.Fatalf("Failed to reconstruct trace: %v", errs)
-	}
+	//got, errs := ReconstructTraces(span13, span14, span11, span21, span12)
```
Comment (songy23) [inline interoptest/src/testcoordinator/validator/trace_constructor_test.go]: nit: can these lines be removed?

File: interoptest/src/testcoordinator/validator/trace_constructor.go [hunk 2]
```diff
@@ -149,7 +150,7 @@ func spanToSimpleSpan(span *tracepb.Span) *SimpleSpan {
 }
 
 func isRoot(span *tracepb.Span) bool {
-	return span.ParentSpanId == nil || len(span.ParentSpanId) == 0
+	return span.ParentSpanId == nil || len(span.ParentSpanId) == 0 || zeroSpan == ToSpanID(span.ParentSpanId)
```
[the remark this answers is not in the document]
Comment (rghetia) [reply interoptest/src/testcoordinator/validator/trace_constructor.go]: parent span id is zero but will the span id ever be zero? That would be invalid span. It should be thrown away during groupSpansByTraceID. what do you think?

File: interoptest/src/testcoordinator/validator/trace_constructor_test.go (hunk 1 above)
Comment (rghetia) [reply interoptest/src/testcoordinator/validator/trace_constructor_test.go]: done.

File: interoptest/src/testcoordinator/validator/trace_constructor.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (songy23) [reply interoptest/src/testcoordinator/validator/trace_constructor.go]: In Java if no implementation is provided, all the spans will have an ID of 0: https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/trace/BlankSpan.java#L41. This is an edge case - it's hard to tell if that is invalid or not, because when integrating OC-Java with other Java libraries, usually no implementation is provided.

Anyway from the perspective of interop tests I think we should always test against a real implementation, so I'm fine with leaving this case as invalid for now.
